dataflow/src/main/java/org/checkerframework/dataflow/cfg/builder/CFGTranslationPhaseOne.java

```diff
@@ -2562,7 +2563,9 @@ private boolean casesAreExhaustive() {
             List<Name> caseLabels = new ArrayList<>(enumConstants.size());
             for (CaseTree caseTree : caseTrees) {
               for (ExpressionTree caseEnumConstant : TreeUtils.caseTreeGetExpressions(caseTree)) {
-                caseLabels.add(((IdentifierTree) caseEnumConstant).getName());
+                if (caseEnumConstant.getKind() != Kind.NULL_LITERAL) {
+                  caseLabels.add(((IdentifierTree) caseEnumConstant).getName());
+                }
               }
             }
             // Could also check that the values match.
```

Comment:
On the line below, now that "null" is a possible value, I think that the cases are exhaustive only if they *also* include "null".  Otherwise, fallthrough is possible.

Reply:
`null` doesn't have to appear for the switch block to be exhaustive.  If a `case null` does not appear and null is the value of the switch selector expressions, then a NPE is thrown.
 
Here's the definition of exhaustive switch blocks: https://docs.oracle.com/javase/specs/jls/se20/preview/specs/patterns-switch-record-patterns-jls.html#jls-14.11.1.1.